scioip34/ABM
# Feature/coopsigsave

In this PR we create the first working prototype of the replay tool with project specific saved files and we define what needs to be saved when the APP_VERSION is CooperativeSignalling.

1. Added meter value and signaling for saving in monitoring/ifdb as new cs specific methods
2. These are called in cs_sims so cs specific attributes are saved
3. These are zarr files and other formats are not implemented
4. These are then loaded back with the DataLoader class that can now load back cs specific zarr files
5. The loaded data is used to instanciate an ExperimentReplay class instance that can visualize what happened in the played experiment (using now cs specific zarr files)
6. The version of the app is controlled with the "APP_VESRION" env flag

To test the method: 
1. create a "test.env" empty file in the root ABM folder (this will be overwritten)
2. run `EXPERIMENT_NAME=test python /path/to/abm/ABM/abm/data/metaprotocol/experiments/archive/CooperativeSignalling/template/ExperimentCoopSigTemplate.py`
3. this will generate experimental data in data/simulation_data/test
4. change the path in ABM/load_data_test.py to the absolute path of the generated test folder
5. the data runs are joined and summarized into a single summary folder under test and a replay window opens up.

Note that most of the functionalities are not yet implemented in the replay tool to handle cs specific stuff yet, only the visualization of the agents.
